### PR TITLE
[xdoctest] reformat example code with google style No.102-104

### PR DESCRIPTION
**Pull request by PommesPeter**
### PR types

Others
### PR changes

Others
### Description

修改如下文件的示例代码，使其通过 `xdoctest` 检查：

- ./python/paddle/nn/functional/adadelta.py
- ./python/paddle/nn/functional/adagrad.py
- ./python/paddle/nn/layer/adam.py

预览：

en: http://preview-paddle-pr-56124.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/abs_en.html
zh: http://preview-paddle-pr-56124.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/abs_cn.html

### 关联 PR：
- #55629
- #55295

@sunzhongkai588 @SigureMo @megemini

### CODE REVIEW

File: python/paddle/optimizer/adadelta.py [hunk 1]
```diff
@@ -70,39 +70,38 @@ class Adadelta(Optimizer):
...
+            >>> linear = paddle.nn.Linear(10, 10)
+            >>> out = linear(inp)
+            >>> loss = paddle.mean(out)
+            >>> beta1 = paddle.to_tensor([0.9], dtype="float32")
+            >>> beta2 = paddle.to_tensor([0.99], dtype="float32")
+            >>> adadelta = paddle.optimizer.Adadelta(learning_rate=0.1, parameters=linear.parameters(), weight_decay=0.01)
+            >>> back = out.backward()
+            >>> adadelta.step()
+            >>> adadelta.clear_grad()
+            >>> # Note that the learning_rate of linear_2 is 0.01.
```
Comment (megemini) [inline python/paddle/optimizer/adadelta.py:86]: 这之前的空行还是保留吧，上下两段代码会比较清晰 ～

File: python/paddle/optimizer/adagrad.py [hunk 2]
```diff
@@ -70,38 +70,37 @@ class Adagrad(Optimizer):
...
+            >>> inp = paddle.rand(shape=[10, 10])
+            >>> linear = paddle.nn.Linear(10, 10)
+            >>> out = linear(inp)
+            >>> loss = paddle.mean(out)
+            >>> adagrad = paddle.optimizer.Adagrad(learning_rate=0.1,
+            ...         parameters=linear.parameters())
+            >>> out.backward()
+            >>> adagrad.step()
+            >>> adagrad.clear_grad()
+            >>> # Note that the learning_rate of linear_2 is 0.01.
```
Comment (megemini) [inline python/paddle/optimizer/adagrad.py:85]: 保留空行

File: python/paddle/optimizer/adam.py [hunk 3]
```diff
@@ -98,63 +98,60 @@ class Adam(Optimizer):
...
+            >>> beta2 = paddle.to_tensor([0.99], dtype="float32")
+            >>> adam = paddle.optimizer.Adam(learning_rate=0.1,
+            ...         parameters=linear.parameters(),
+            ...         beta1=beta1,
+            ...         beta2=beta2,
+            ...         weight_decay=0.01)
+            >>> loss.backward()
+            >>> adam.step()
+            >>> adam.clear_grad()
+            >>> # Note that the learning_rate of linear_2 is 0.01.
```
Comment (megemini) [inline python/paddle/optimizer/adam.py:134]: 保留空行

**Review by SigureMo — DISMISSED** [review]
LGTMeow 🐾

**Review by megemini — COMMENTED** [review]
挺好，没啥大问题 ～

@SigureMo @sunzhongkai588 `Adagrad` 的中文文档里面，缺少了公共方法的描述～

英文
image

中文
image

File: python/paddle/optimizer/adadelta.py (hunk 1 above)
Comment (PommesPeter) [reply python/paddle/optimizer/adadelta.py:86]: 好的，done

File: python/paddle/optimizer/adagrad.py (hunk 2 above)
Comment (PommesPeter) [reply python/paddle/optimizer/adagrad.py:85]: done

File: python/paddle/optimizer/adam.py (hunk 3 above)
Comment (PommesPeter) [reply python/paddle/optimizer/adam.py:134]: done

**Comment by PommesPeter** [comment]
已修改~

**Comment by SigureMo** [comment]
codestyle 流水线需要看一下

**Comment by PommesPeter** [comment]
> codestyle 流水线需要看一下

已修复~

File: python/paddle/optimizer/lr.py [hunk 4]
```diff
@@ -128,29 +128,29 @@ def step(self, epoch=None):
...
-                value = paddle.arange(26, dtype='float32')
-                a = paddle.reshape(value, [2, 13])
-                linear = paddle.nn.Linear(13, 5)
-                adadelta = paddle.optimizer.Adadelta(learning_rate=0.0003, epsilon=1e-06, rho=0.95,
-                                            parameters = linear.parameters())
-                out = linear(a)
-                out.backward()
-                adadelta.step()
-                adadelta.clear_grad()
+                >>> import paddle
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

                >>> import paddle
```

image

File: python/paddle/optimizer/lr.py [hunk 5]
```diff
@@ -64,31 +64,31 @@ class LRScheduler:
...
+            ...             raise ValueError('gamma should be < 1.0.')
+            ...
+            ...         self.step_size = step_size
+            ...         self.gamma = gamma
+            ...         super().__init__(learning_rate, last_epoch, verbose)
+            ...
+            ...     def get_lr(self):
+            ...         i = self.last_epoch // self.step_size
+            ...         return self.base_lr * (self.gamma**i)
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py [hunk 6]
```diff
@@ -267,57 +267,59 @@ class NoamDecay(LRScheduler):
...
+            ...     for batch_id in range(5):
+            ...         x = paddle.uniform([10, 10])
+            ...         out = linear(x)
+            ...         loss = paddle.mean(out)
+            ...         loss.backward()
+            ...         sgd.step()
+            ...         sgd.clear_gradients()
+            ...         scheduler.step()    # If you update learning rate each step
+            ...     # scheduler.step()        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 7]
```diff
@@ -267,57 +267,59 @@ class NoamDecay(LRScheduler):
...
+            ...         out = exe.run(
+            ...             main_prog,
+            ...             feed={
+            ...                 'x': np.random.randn(3, 4, 5).astype('float32'),
+            ...                 'y': np.random.randn(3, 4, 5).astype('float32')
+            ...             },
+            ...             fetch_list=loss.name)
+            ...         scheduler.step()    # If you update learning rate each step
+            ...     # scheduler.step()        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py [hunk 8]
```diff
@@ -377,56 +379,58 @@ class PiecewiseDecay(LRScheduler):
...
+            ...     for batch_id in range(5):
+            ...         x = paddle.uniform([10, 10])
+            ...         out = linear(x)
+            ...         loss = paddle.mean(out)
+            ...         loss.backward()
+            ...         sgd.step()
+            ...         sgd.clear_gradients()
+            ...         scheduler.step()    # If you update learning rate each step
+            ...     # scheduler.step()        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py [hunk 9]
```diff
@@ -474,54 +478,55 @@ class NaturalExpDecay(LRScheduler):
...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 10]
```diff
@@ -561,57 +566,57 @@ class InverseTimeDecay(LRScheduler):
...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 11]
```diff
@@ -561,57 +566,57 @@ class InverseTimeDecay(LRScheduler):
...
+            ...         out = exe.run(
+            ...             main_prog,
+            ...             feed={
+            ...                 'x': np.random.randn(3, 4, 5).astype('float32'),
+            ...                 'y': np.random.randn(3, 4, 5).astype('float32')
+            ...             },
+            ...             fetch_list=loss.name)
+            ...         scheduler.step()    # If you update learning rate each step
+            ...     # scheduler.step()        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py [hunk 12]
```diff
@@ -664,56 +669,56 @@ class PolynomialDecay(LRScheduler):
...
+            ...     for batch_id in range(5):
+            ...         x = paddle.uniform([10, 10])
+            ...         out = linear(x)
+            ...         loss = paddle.mean(out)
+            ...         loss.backward()
+            ...         sgd.step()
+            ...         sgd.clear_gradients()
+            ...         scheduler.step()    # If you update learning rate each step
+            ...     # scheduler.step()        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 13]
```diff
@@ -795,58 +800,58 @@ class LinearWarmup(LRScheduler):
...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 14]
```diff
@@ -935,56 +940,56 @@ class ExponentialDecay(LRScheduler):
...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 15]
```diff
@@ -1033,56 +1038,56 @@ class MultiStepDecay(LRScheduler):
...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 16]
```diff
@@ -1147,56 +1152,56 @@ class StepDecay(LRScheduler):
...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 17]
```diff
@@ -1251,57 +1256,57 @@ class LambdaDecay(LRScheduler):
...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 18]
```diff
@@ -1251,57 +1256,57 @@ class LambdaDecay(LRScheduler):
...
+            ...         out = exe.run(
+            ...             main_prog,
+            ...             feed={
+            ...                 'x': np.random.randn(3, 4, 5).astype('float32'),
+            ...                 'y': np.random.randn(3, 4, 5).astype('float32')
+            ...             },
+            ...             fetch_list=loss.name)
+            ...         scheduler.step()    # If you update learning rate each step
+            ...     # scheduler.step()        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py [hunk 19]
```diff
@@ -1359,57 +1364,57 @@ class ReduceOnPlateau(LRScheduler):
...
+            ...     for batch_id in range(5):
+            ...         x = paddle.uniform([10, 10])
+            ...         out = linear(x)
+            ...         loss = paddle.mean(out)
+            ...         loss.backward()
+            ...         sgd.step()
+            ...         sgd.clear_gradients()
+            ...         scheduler.step(loss)    # If you update learning rate each step
+            ...     # scheduler.step(loss)        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 20]
```diff
@@ -1359,57 +1364,57 @@ class ReduceOnPlateau(LRScheduler):
...
+            ...         out = exe.run(
+            ...             main_prog,
+            ...             feed={
+            ...                 'x': np.random.randn(3, 4, 5).astype('float32'),
+            ...                 'y': np.random.randn(3, 4, 5).astype('float32')
+            ...             },
+            ...             fetch_list=loss.name)
+            ...         scheduler.step(out[0])    # If you update learning rate each step
+            ...     # scheduler.step(out[0])        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py [hunk 21]
```diff
@@ -1590,56 +1595,56 @@ class CosineAnnealingDecay(LRScheduler):
...
+            ...     for batch_id in range(5):
+            ...         x = paddle.uniform([10, 10])
+            ...         out = linear(x)
+            ...         loss = paddle.mean(out)
+            ...         loss.backward()
+            ...         sgd.step()
+            ...         sgd.clear_gradients()
+            ...         scheduler.step()    # If you update learning rate each step
+            ...     # scheduler.step()        # If you update learning rate each epoch
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 22]
```diff
@@ -1714,23 +1719,23 @@ class MultiplicativeDecay(LRScheduler):
...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py [hunk 23]
```diff
@@ -1795,55 +1800,55 @@ class OneCycleLR(LRScheduler):
...
+            >>> for epoch in range(5):
+            ...     for batch_id in range(20):
+            ...         x = paddle.uniform([10, 10])
+            ...         out = linear(x)
+            ...         loss = paddle.mean(out)
+            ...         loss.backward()
+            ...         sgd.step()
+            ...         sgd.clear_gradients()
+            ...         scheduler.step()        # You should update learning rate each step
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion

```

File: python/paddle/optimizer/lr.py [hunk 24]
```diff
@@ -1795,55 +1800,55 @@ class OneCycleLR(LRScheduler):
...
+            ...     for batch_id in range(20):
+            ...         out = exe.run(
+            ...             main_prog,
+            ...             feed={
+            ...                 'x': np.random.randn(3, 4, 5).astype('float32'),
+            ...                 'y': np.random.randn(3, 4, 5).astype('float32')
+            ...             },
+            ...             fetch_list=loss.name)
+            ...         scheduler.step()    # You should update learning rate each step
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py [hunk 25]
```diff
@@ -2045,55 +2050,55 @@ class CyclicLR(LRScheduler):
...
+            >>> for epoch in range(5):
+            ...     for batch_id in range(20):
+            ...         x = paddle.uniform([10, 10])
+            ...         out = linear(x)
+            ...         loss = paddle.mean(out)
+            ...         loss.backward()
+            ...         sgd.step()
+            ...         sgd.clear_gradients()
+            ...         scheduler.step()        # You should update learning rate each step
+            ...
```
Comment (SigureMo) [inline python/paddle/optimizer/lr.py]:
```suggestion
```

File: python/paddle/optimizer/lr.py
Comment (SigureMo) [inline python/paddle/optimizer/lr.py:1]: 这个文件是？和人家 #56225 冲突了哈，我说怎么 review 了两遍 lr

以认领为准吧……

Comment (PommesPeter) [reply python/paddle/optimizer/lr.py:1]: 抱歉，不小心把这个改动加进来了，已删除。

File: python/paddle/optimizer/adadelta.py [hunk 26]
```diff
@@ -70,39 +70,40 @@ class Adadelta(Optimizer):
...
-                    'params': linear_2.parameters(),
-                    'weight_decay': 0.001,
-                    'learning_rate': 0.1,
-                }],
-                weight_decay=0.01)
-            out.backward()
-            adadelta.step()
-            adadelta.clear_grad()
+            >>> import paddle
+            >>> paddle.seed(2023)
```
Comment (SigureMo) [inline python/paddle/optimizer/adadelta.py]: 这个 seed 是用来干啥的，这里也没有输出需要保证啊

非必要不要加 seed，下同

File: python/paddle/optimizer/adam.py (hunk 3 above)
Comment (SigureMo) [reply python/paddle/optimizer/adam.py:134]: 这里是不是忘加了

Comment (PommesPeter) [reply python/paddle/optimizer/adam.py:134]: 已修改

File: python/paddle/optimizer/adadelta.py (hunk 26 above)
Comment (PommesPeter) [reply python/paddle/optimizer/adadelta.py]: 之前的样例提到涉及随机操作加上 seed

Comment (SigureMo) [reply python/paddle/optimizer/adadelta.py]: 随机操作不影响输出的不需要加，能不加就不加吧

@megemini 觉得呢？

Comment (megemini) [reply python/paddle/optimizer/adadelta.py]: @PommesPeter 非常抱歉，是我表达的不够清楚 ～

加 seed 是为了保证输出的固定，以方便利用 doctest 等工具进行检查，也就是对比程序执行的输出，与示例中的输出是否一致。

这段代码没有看到输出的部分，为了尽量保持代码的简洁，应该就不需要加 seed 了，请悉知！ ：）

Comment (PommesPeter) [reply python/paddle/optimizer/adadelta.py]: 好的，明白了，已删除。

**Review by megemini — APPROVED** [review]
LGTM ～

@sunzhongkai588  Adagrad 的中文文档里面，缺少了公共方法的描述，其他 OK ～

**Comment by sunzhongkai588** [comment]
> @sunzhongkai588 Adagrad 的中文文档里面，缺少了公共方法的描述，其他 OK ～

@SigureMo  公共方法中文应该不需要补充吧？英文是自动抽取公共方法的docstring。（话说英文这方面..这应该算bug么？可以当成任务发出来修么 🥹）

**Comment by SigureMo** [comment]
> @SigureMo 公共方法中文应该不需要补充吧？英文是自动抽取公共方法的docstring。（话说英文这方面..这应该算bug么？可以当成任务发出来修么 🥹）

应该只是中英文生成工具不一致的问题吧，公共方法是自动继承的么？如果是自动继承的话我记得可以配置啥 no-inherit 啥的，好像是在 gen_doc.py 里

**Review by sunzhongkai588 — APPROVED** [review]
LGTM

**Comment by sunzhongkai588** [comment]
> > @SigureMo 公共方法中文应该不需要补充吧？英文是自动抽取公共方法的docstring。（话说英文这方面..这应该算bug么？可以当成任务发出来修么 🥹）
> 
> 应该只是中英文生成工具不一致的问题吧，公共方法是自动继承的么？如果是自动继承的话我记得可以配置啥 no-inherit 啥的，好像是在 gen_doc.py 里

是自动继承的好像。而且 gen_doc.py 大概率又要处于没人维护的状态..anyway吧先这样
